rllib/utils/torch_utils.py

```diff
@@ -66,12 +67,15 @@ def apply_grad_clipping(
                 global_norm = global_norm.cpu().numpy()
 
             grad_gnorm += min(global_norm, clip_value)
+        else:
+            num_none_grads += 1
 
-    if grad_gnorm > 0:
-        return {"grad_gnorm": grad_gnorm}
-    else:
+    # Note (Kourosh): grads could indeed be zero. This method should still return
```

Comment:
Good catch. Dumb question, though: If all grads are zero (and thus grad_norm is zero), isn't that a strong sign that something is wrong with the loss' dependency on the model params?

Reply:
Nope, I don't exactly recall what happened here that I needed to do this special handling but in general, you can have scenarios that at some iteration or initialization you are doing a dry-run on fake data which happens to to get on a relu activation at the very final layer. For that batch your gradient could end up being zero and you should not barf. Next batch things will be fine.

Reply:
The caller should handle whether zero grad_norm is an error or no. This function is supposed to behave differently between grad_norm of zero vs. `num_none_grads = len(optimizer.param_groups)`